homeassistant/components/fan/alfawise.py

```diff
@@ -0,0 +1,109 @@
+"""
+Alfawise fan platform that has a fake fan.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/fan.alfawise/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.fan import (SPEED_LOW, SPEED_HIGH,
+                                          FanEntity, SUPPORT_SET_SPEED)
+from homeassistant.const import CONF_DEVICES, CONF_NAME, CONF_MAC
+from homeassistant.const import STATE_OFF
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.exceptions import PlatformNotReady
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pyAlfawise==0.4-beta']
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Required(CONF_MAC): cv.string,
+    vol.Required(CONF_NAME): cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Required(CONF_DEVICES): {cv.string: DEVICE_SCHEMA}})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Alfawise fan platform."""
+    import pyAlfawise
+    fans = []
+    for ipaddr, device_config in config[CONF_DEVICES].items():
+        try:
+            name = device_config[CONF_NAME]
+            macaddr = device_config[CONF_MAC]
+            _LOGGER.debug("Adding configured %s", name)
+            device = pyAlfawise.Alfawise(macaddr, ipaddr)
+            fan = AlfawiseFan(hass, device, name)
+            fans.append(fan)
+        except Exception:
```

Comment:
The python library the component is based on has no specific exception in case the device can not be reached. Should I open an issue or made a PR for the library before to fix this ?

Reply:
It would be best if the library implemented its own exceptions and raised them as needed while allowing users to import those to catch the exceptions.